### PR TITLE
[SIMS-#925][API][Web][Workflow] Program Information Request application error message

### PR DESCRIPTION
**Pull request by henryoforeh-aot**
This implementation was done to show the appropriate error label and message from the Api response.

### CODE REVIEW

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue [hunk 1]
```diff
@@ -213,17 +220,21 @@ export default {
...
-          if (error.includes(customError)) {
-            errorMsg = error.replace(customError, "").trim();
-          }
-        });
+          APPLICATION_DATE_OVERLAP_ERROR,
+        ].some(
+          (errorCode: string) => errorCode === error.response.data?.errorType,
+        );
+        if (isCustomError) {
+          errorLabel = "Application Overlap";
```
Comment (ann-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:234]: mm, according to 231 logic, all errors mentioned in the list i.e 226, will enter here right? or am I missing anything

Comment (ann-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:234]: and for `INVALID_STUDY_DATES,
          OFFERING_START_DATE_ERROR,
          PIR_OR_DATE_OVERLAP_ERROR,` errors also the label will be `errorLabel = "Application Overlap";`

File: sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts [hunk 2]
```diff
@@ -360,7 +361,10 @@ export class ProgramInfoRequestController extends BaseController {
         ].includes(error.name)
       ) {
         throw new UnprocessableEntityException(
-          `${error.name} ${error.message}`,
+          new ApiProcessError(
+            `${error.name} ${error.message}`,
```
Comment (dheepak-aot) [inline sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:365]: error.message is alone sufficient here. not required to concatenate with error.name.

File: sources/packages/api/src/route-controllers/application/application.students.controller.ts [hunk 3]
```diff
@@ -189,7 +189,10 @@ export class ApplicationStudentsController extends BaseController {
       );
       if (notValidDates) {
         throw new UnprocessableEntityException(
-          `${INVALID_STUDY_DATES} ${notValidDates}`,
+          new ApiProcessError(
+            `${INVALID_STUDY_DATES} ${notValidDates}`,
```
Comment (dheepak-aot) [inline sources/packages/api/src/route-controllers/application/application.students.controller.ts:193]: INVALID_STUDY_DATES is the error code. 
It has to be used like this

`throw new UnprocessableEntityException(
          new ApiProcessError(
            notValidDates,
            INVALID_STUDY_DATES,
          ),
        );`

Comment (ann-aot) [inline sources/packages/api/src/route-controllers/application/application.students.controller.ts:193]: `INVALID_STUDY_DATES` and  `APPLICATION_DATE_OVERLAP_ERROR` are different right? I believe `INVALID_STUDY_DATES` is when a student request a PIR and the selected study period difference is not in between 42 days and 365 days. the controller will throw this exception

File: sources/packages/web/src/views/institution/locations/program-info-request/LocationEditProgramInfoRequest.vue [hunk 4]
```diff
@@ -176,18 +176,22 @@ export default {
...
-            errorMsg = error.replace(customError, "").trim();
-          }
-        });
+          APPLICATION_DATE_OVERLAP_ERROR,
+        ].some(
+          (errorCode: string) => errorCode === error.response.data?.errorType,
+        );
+
+        if (isCustomError) {
+          errorLabel = "Application Overlap";
```
Comment (dheepak-aot) [inline sources/packages/web/src/views/institution/locations/program-info-request/LocationEditProgramInfoRequest.vue:192]: I forgot to mention this point when we spoke. The error label must be Generic to all these custom errors. 
OFFERING_START_DATE_ERROR,
          INVALID_STUDY_DATES,
          OFFERING_INTENSITY_MISMATCH,
          APPLICATION_DATE_OVERLAP_ERROR,

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue (hunk 1 above)
Comment (dheepak-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:234]: Same here. Error label must be generic to all the custom errors.

Comment (ann-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:234]: same in file `sources/packages/web/src/views/institution/locations/program-info-request/LocationEditProgramInfoRequest.vue`

File: sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts [hunk 5]
```diff
@@ -360,7 +361,10 @@ export class ProgramInfoRequestController extends BaseController {
...
+            APPLICATION_DATE_OVERLAP_ERROR,
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts:366]: I am not understanding this

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the adjustments. Looks good 👍

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the changes @henryoforeh-aot 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @henryoforeh-aot

**Pull request merged**